### PR TITLE
Avoid the unsafe getenv function

### PR DESCRIPTION
**Pull request by GrahamCampbell**
#### TLDR: `getenv` is not thread safe!

Obviously this doesn't matter when this library is using with composer running in the console, but it does matter when this code is running on a web server. The `genv` and `putenv` functions are not threadsafe, so environment variables set in one thread using `putenv` may leak into another via `getenv`.

This not just a "theoretical" bug. It actually happens in real apps and serves, and was the result of hundreds of bug reports on the Laravel framework over the years, where people were confused by variables not being what they expected!

---

EDIT: In order to minimize breakage, for CLI apps, I've allowed to still read from `getenv` for that.

### CODE REVIEW

**Comment by brunogaspar** [comment]
Not sure i really want to change this on version 2, although, I've removed this for the upcoming version 3.

Will think about it.

Thanks!

**Comment by GrahamCampbell** [comment]
I'll email you privately some comments out this.

File: src/Config.php [hunk 1]
```diff
@@ -76,9 +76,28 @@ public function __construct($version, $apiKey, $apiVersion)
...
+     * @return string|null
+     */
+    private static function getEnvVariable($name, $default = null)
+    {
+        if (isset($_SERVER[$name])) {
+            return $_SERVER[$name];
+        }
+
+        if (PHP_SAPI === 'cli' && ($value = getenv($name) !== false)) {
+            return $value;
```
Comment (brunogaspar) [inline src/Config.php]: This returns a boolean, since the condition above is performing a check after the assignment.

It seems the closing `)` needs to be earlier

```php
if (PHP_SAPI === 'cli' && ($value = getenv($name)) !== false) {
```

want to do this change or do it a tad differently?

Comment (GrahamCampbell) [reply src/Config.php]: Yes, I just noticed this too! Fixed. :)

**Comment by brunogaspar** [comment]
Looks good, i think only one test, related to this at least, will fail, but it's easy to fix.

Going to merge and then i'll tweak the test.

**Comment by GrahamCampbell** [comment]
Thanks! :)

**Comment by brunogaspar** [comment]
Thank you for the fix.

Will push to 2.4 and do tags on both versions a bit later.
